ros2_control_test_assets/include/ros2_control_test_assets/descriptions.hpp

```diff
@@ -453,10 +579,440 @@ const auto diffbot_urdf =
+    </joint>
+  </ros2_control>
+  )";
+
+const auto gripper_hardware_resources_mimic_true_no_command_if =
+  R"(
+  <ros2_control name="TestGripper" type="system">
+    <joint name="right_finger_joint">
+      <command_interface name="effort"/>
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+      <state_interface name="effort"/>
+    </joint>
+    <joint name="left_finger_joint" mimic="true">
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+    </joint>
+  </ros2_control>
+  )";
+
+// TODO(christophfroehlich) delete deprecated config test
+const auto gripper_hardware_resources_mimic_deprecated =
+  R"(
+  <ros2_control name="TestGripper" type="system">
+    <joint name="right_finger_joint">
+      <command_interface name="effort"/>
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+      <state_interface name="effort"/>
+    </joint>
+    <joint name="left_finger_joint">
+      <param name="mimic">right_finger_joint</param>
+      <param name="multiplier">2</param>
+      <param name="offset">1</param>
+      <command_interface name="position"/>
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+    </joint>
+  </ros2_control>
+  )";
+
+const auto gripper_hardware_resources_mimic_deprecated_unknown_joint =
+  R"(
+  <ros2_control name="TestGripper" type="system">
+    <joint name="right_finger_joint">
+      <command_interface name="effort"/>
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+      <state_interface name="effort"/>
+    </joint>
+    <joint name="left_finger_joint">
+      <param name="mimic">middle_finger_joint</param>
+      <param name="multiplier">1</param>
+      <command_interface name="position"/>
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+    </joint>
+  </ros2_control>
+  )";
+// end delete deprecated config test
+
+const auto gripper_hardware_resources_mimic_true_command_if =
+  R"(
+  <ros2_control name="TestGripper" type="system">
+    <joint name="right_finger_joint">
+      <command_interface name="effort"/>
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+      <state_interface name="effort"/>
+    </joint>
+    <joint name="left_finger_joint" mimic="true">
+      <command_interface name="effort"/>
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+    </joint>
+  </ros2_control>
+  )";
+
+const auto gripper_hardware_resources_mimic_false_command_if =
+  R"(
+  <ros2_control name="TestGripper" type="system">
+    <joint name="right_finger_joint">
+      <command_interface name="effort"/>
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+      <state_interface name="effort"/>
+    </joint>
+    <joint name="left_finger_joint" mimic="false">
+      <command_interface name="effort"/>
```

Comment:
I'm not sure about this case, but here it might make sense as you are not mimicking the parent joint

Reply:
this is used to check if it does not add a mimic joint from the URDF in case the attribute `mimic=false` is set, and does not throw an error. 
There is another test throwing an error if mimic=true but a command_interface is set.

Reply:
here one might deactivate the mimic on purpose but claim the command_interface

Reply:
Sure, I agree. If they deactivate the mimic behaviour then having a command interface can be allowed